monitoring/mongodb/alerts.yaml

```
@@ -139,3 +139,66 @@ groups:
     annotations:
       description: 'MongoDb has low disk space'
       summary: 'MongoDb has low disk space'
+  
+  - alert: MongoDbPodRecovering
+    expr: |
+      avg_over_time(mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", rs_state="3"}[1h]) == 3
+    for: 1h
```

this alert triggers after 2 hours : the expression becomes true after 1h, then we "ignore" the error for 1h.

average over time can only match if the value did not change (and not sure how it handles "missing" datapoints), and make computation slower...

The following will do the same more efficiently, just looking at the current state every minute or so ; and alert manager will actually fire the alert only if every point in the `for` duration (1h) matches the alert : which is exactly what you wanted to do, right?

```suggestion
      mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", rs_state="3"} == 3
    for: 1h
```